src/FSharpVersionUtils/semVer.fs

```diff
@@ -0,0 +1,160 @@
+﻿namespace datNET
+
+module SemVer =
+
+  type SemVer =
+    {
+      Major : int
+      Minor : int
+      Patch : int
+      Pre   : string option
+      Meta  : string option
+    }
+
+  let stringify semVer =
+    let baseStr =
+      [| semVer.Major ; semVer.Minor ; semVer.Patch |]
+      |> Seq.map (fun i -> i.ToString())
+      |> String.concat "."
+
+    let pre =
+      match semVer.Pre with
+      | Some v -> sprintf "-%s" v
+      | None   -> ""
+
+    let meta =
+      match semVer.Meta with
+      | Some v -> sprintf "+%s" v
+      | None   -> ""
+
+    String.concat "" [| baseStr ; pre ; meta |]
+
+  let private _ParseFail str =
+    let message = sprintf "Invalid semantic version string: %s" str
+    raise (new System.FormatException(message))
+
+  (* The following might be better done as regexes *)
+  let private _headAndMeta (str: string) =
```

Comment:
From verbal discussion we agreed on the following rules

* private values are prepended by an underscore
* public values are not
* all values are camelCased